nbd/src/main.cpp

```diff
@@ -95,6 +97,112 @@ static void Usage() {
         << std::endl;
 }
 
+// use for record image and device info to auto map when boot
+static void AddRecord() {
+    std::fstream fs;
+    int fd;
+    fs.open(CURVETAB_PATH, std::fstream::in);
+    if (!fs.is_open()) {
+        std::cerr << "curve-nbd: open curvetab file failed.";
+        return;
+    }
+    fd = static_cast<__gnu_cxx::stdio_filebuf< char > * const >
+                                            (fs.rdbuf())->fd();
+    if (0 == flock(fd, LOCK_SH )) {
+        // check if the record exists
+        std::string lineString;
+        while (getline(fs, lineString)) {
+            if (!lineString.empty() && lineString[0] != '#') {
+                std::vector<std::string> lineArray;
+                std::stringstream ss(lineString);
+                std::string str;
+                while (getline(ss, str, '\t')) {
+                    lineArray.push_back(str);
+                }
+                if (lineArray.size() != 2 && lineArray.size() != 3) {
+                    std::cerr << "curve-nbd: curvetab file content error.";
+                    fs.close();
```

Comment:
> The `close()` is redunrant, because the file stream's destructor will close the file.

but this fs reused latter in this function. so I think close is needed